### PR TITLE
Serial driver

### PR DESCRIPTION
**Pull request by RoryHemmings**
Add basic serial driver
- Convert all outbb to outb
- outb is now: port, value

### CODE REVIEW

**Review by danieltherealyang — APPROVED** [review]
looks legit

File: source/kernel/serial.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    return 0;
+}
+
+inline static int transmit_buffer_empty(enum COM_PORT port)
+{
+    /* Read from transmit buffer empty register */
+    return inb(port + 5) & 0x20;
+}
+
+inline static int putc_serial(enum COM_PORT port, char c)
```
Comment (r2dev2) [inline source/kernel/serial.c:52]: Returns an int in function signature but i don't see a return statement, can you make it return the error code and also propagate it to the main write_serial()

File: source/kernel/serial.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    // If test byte was not returned, serial is faulty
+    if (inb(port + 0) != 0xAE)
+      return 0;
+
+    // // Otherwise, serial is operational, therefore disable loopback mode
+    outb(port + 4, 0x0F);
+    return 1;
+}
+
+void write_serial(enum COM_PORT port, const char *s)
```
Comment (r2dev2) [inline source/kernel/serial.c:35]: Are we moving away from our fat string and to char *?

**Review by r2dev2 — CHANGES_REQUESTED** [review]
Just that minor thing about function return

File: source/kernel/serial.c (hunk 2 above)
Comment (RoryHemmings) [reply source/kernel/serial.c:35]: Yeah, Daniel got rid of it in the pic branch. We agreed to get rid of it entirely yesterday.

File: source/kernel/serial.c (hunk 1 above)
Comment (RoryHemmings) [reply source/kernel/serial.c:52]: I'll fix this in our update branch, we still have some more things to add to the driver as well.

**Pull request merged**